staging/src/k8s.io/apiserver/pkg/storage/tests/cacher_test.go

```diff
@@ -796,6 +823,7 @@ func testSetup(t *testing.T, opts ...setupOption) (context.Context, *cacherstora
 		GetAttrsFunc:   GetPodAttrs,
 		NewFunc:        newPod,
 		NewListFunc:    newPodList,
+		IndexerFuncs:   triggerFuncs,
```

Comment:
instead of this, let's create an explicit

```
func withSpecNodeNameIndexerFuncs(options *setupOptions) {
  opts.indexerFuncs = map[string]storage.IndexerFunc{
    "spec.nodeName": func (obj runtime.Object) string {
	pod, ok := obj.(*example.Pod)
	if !ok {
		return ""
	}
	return pod.Spec.NodeName
  }
}
```
2) add appropriate field to setupOptions
3) remove the file-level funcs/vars you created

and use that explicitly in your tests